game-core/src/main/java/games/strategy/triplea/attachments/TriggerAttachment.java

```diff
@@ -1508,8 +1508,18 @@ private static void placeUnits(
     final boolean clearFirst = clearFirstNewValue.getFirst();
     final String newValue = clearFirstNewValue.getSecond();
 
-    final boolean isValueTheSame =
-        newValue.equals(propertyAttachment.getRawPropertyString(propertyName));
+    final boolean isValueTheSame;
+    try {
+      isValueTheSame = newValue.equals(propertyAttachment.getRawPropertyString(propertyName));
+    } catch (final UnsupportedOperationException e) {
+      throw new UnsupportedOperationException(
```

Comment:
The extra debug information here is excellent - much better error handling  :+1: